src/scripts/train/OLMo3/OLMo3-32B-midtraining.py

```diff
@@ -0,0 +1,129 @@
+from olmo_core.optim.scheduler import LinearWithWarmup, SchedulerUnits
+from olmo_core.train import Duration
+from olmo_core.train.train_module import TransformerTrainModuleConfig
+
+from olmo_core.nn.attention import SlidingWindowAttentionConfig
+
+from olmo_core.data import InstanceFilterConfig
+
+SEQ_LENGTH = 8192
+GLOBAL_BATCH_SIZE = 4 * 1024 * 1024  # ~4M tokens
+MAX_TOKENS = 100_000_000_000  # 100B
+LR = 0.00020712352850360292 #TO BE UPDATED
+SEED = 1337
+
+
+def build_experiment_config(cli_context: CliContext) -> ExperimentConfig:
+    run_name_with_ts = (
+        f"{cli_context.run_name}-{datetime.now().astimezone().strftime('%Y%m%dT%H%M%S%z')}"
+    )
+    root_dir = get_root_dir(cli_context.cluster)
+    work_dir = get_work_dir(root_dir)
+    save_dir = f"{root_dir}/checkpoints/{run_name_with_ts}"
+
+    beaker_launch_config: Optional[BeakerLaunchConfig] = build_launch_config(
+        name=cli_context.run_name,
+        cmd=cli_context.remote_cmd,
+        cluster=cli_context.cluster,
+        root_dir=root_dir,
+        workspace="ai2/olmo-3-microanneals",
+        num_nodes=64,
+        nccl_debug=False,
+        # override priority from the CLI eg `--launch.priority=high`
+    )
+
+    tokenizer_config = TokenizerConfig.dolma2()
+    # model_config = TransformerConfig.olmo3_7B(vocab_size=tokenizer_config.padded_vocab_size())
+    model_config = TransformerConfig.olmo2_32B(vocab_size=tokenizer_config.padded_vocab_size())
+    model_config.block.attention.sliding_window = SlidingWindowAttentionConfig(
+        force_full_attention_on_first_layer=False,
+        force_full_attention_on_last_layer=True,
+        pattern=[4096, 4096, 4096, -1],
+    )
+    model_config.block.attention.use_flash = True
+
+    train_module_config: TransformerTrainModuleConfig = cookbook.configure_train_module(
+        max_sequence_length=SEQ_LENGTH,
+        rank_microbatch_size=SEQ_LENGTH,
+        learning_rate=LR,
+        scheduler=LinearWithWarmup(units=SchedulerUnits.steps, warmup=0, alpha_f=0.0),
+        activation_memory_budget=0.5,
+        dp_shard_degree=64,
+    )
+
+    source_list = SourceMixtureList.from_yaml(
+        "src/olmo_core/data/source_mixtures/OLMo3-32B-midtraining-modelnamefilter.yaml"
+    )
+    source_list.validate()
+    dataset_config = NumpyFSLDatasetConfig.from_src_mix(
+        src_mix=SourceMixtureDatasetConfig(
+            source_list=source_list,
+            requested_tokens=MAX_TOKENS,
+            global_batch_size=GLOBAL_BATCH_SIZE,
+            processes=16,
+            seed=SEED,
+        ),
+        tokenizer=tokenizer_config,
+        work_dir=work_dir,
+        sequence_length=SEQ_LENGTH,
+        instance_filter_config=InstanceFilterConfig(
+                repetition_max_period=13, repetition_min_period=1, repetition_max_count=32
+                )
+    )
+
+    data_loader_config = NumpyDataLoaderConfig(
+        global_batch_size=GLOBAL_BATCH_SIZE, seed=SEED, num_workers=4
```

Comment:
We'll have to set the seed twice on the command line, or fix #447 .

Reply:
@tyler-romero is fixing #447, at least in this branch.

Reply:
https://github.com/allenai/OLMo-core/pull/449